pageserver/src/tenant/timeline/compaction.rs

```diff
@@ -1137,6 +1125,10 @@ impl Timeline {
 
             if !self.shard_identity.is_key_disposable(&key) {
                 if writer.is_none() {
+                    if self.cancel.is_cancelled() {
```

Comment:
this cancel will never be hit if at least one delta layer is produced? probably need to move out of the if branch

Reply:
Hmm weird, I wonder if this was ever needed then..

Though... I was thinking this would be how we open the first file?

Reply:
Isn't this the only way we the L1 files for writing?

Reply:
okay I didn't see the code that set writer to None, so this branch is hit every time we finished a new layer and want to start a new layer

so the expectation here is that we check the cancel condition on switching the writer, instead of immediately doing that (i.e., for each of the iteration we visit a key)?

Reply:
> so the expectation here is that we check the cancel condition on switching the writer, instead of immediately doing that (i.e., for each of the iteration we visit a key)?

that is the implementation. the thinking is that it's very fast to feed the writer and so it would just be unnecessary contention on the cancellation token to check more often.